src/NuGet.Clients/NuGet.PackageManagement.UI/Actions/UIActionEngine.cs

```diff
@@ -986,7 +990,8 @@ private PackageLicenseInfo GeneratePackageLicenseInfo(IPackageSearchMetadata met
                         deleted.Add(new AccessiblePackageIdentity(uninstalled[packageId]));
                     }
                 }
-
+                // Everything added which didn't already have a source mapping will be mentioned in the Preview Window.
+                GetNewSourceMappingsFromAddedPackages(ref newSourceMappings, userAction, added, uiService.UIContext.PackageSourceMapping);
```

Comment:
The source mapping object, is it the update one? Or is this the original-pre operation one?

Reply:
It's the original source mappings, because the current design is a provider which is flagged to not save changes to disk during preview restore. 
https://github.com/NuGet/NuGet.Client/blob/77d0b0f78ae21813fc605fef3e2c62f92ae84a7e/src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs#L2916

Then after preview restore, it saves mappings once restore completes:
https://github.com/NuGet/NuGet.Client/blob/77d0b0f78ae21813fc605fef3e2c62f92ae84a7e/src/NuGet.Clients/NuGet.PackageManagement.UI/Actions/UIActionEngine.cs#L469